components/Modal/TokenAuctionBidModal.js

```diff
@@ -0,0 +1,295 @@
+				args: { account_id: currentUser },
+			})
+
+			const usedStorage = JSBI.multiply(
+				JSBI.BigInt(parseInt(supplyPerOwner) + 1),
+				JSBI.BigInt(STORAGE_ADD_MARKET_FEE)
+			)
+
+			if (JSBI.greaterThanOrEqual(JSBI.BigInt(currentStorage), usedStorage)) {
+				return true
+			}
+			return false
+		} catch (err) {
+			sentryCaptureException(err)
+		}
+	}
+
+	const onPlaceBidAuction = async ({ bidAuctionAmount }) => {
+		setIsBidding(true)
+
+		trackUpdateListingToken(data.token_id)
+
+		const hasDepositStorage = await hasStorageBalance()
+
+		try {
+			const depositParams = { receiver_id: currentUser }
+
+			const params = {
+				nft_contract_id: data.contract_id,
+				...(data.token_id
+					? { token_id: data.token_id }
+					: { token_series_id: data.token_series_id }),
+				ft_token_id: 'near',
+				amount: parseNearAmount(bidAuctionAmount),
+			}
+
+			let res
+			if (hasDepositStorage) {
+				res = await WalletHelper.signAndSendTransaction({
+					receiverId: process.env.MARKETPLACE_CONTRACT_ID,
+					actions: [
+						{
+							methodName: 'add_bid',
+							args: params,
+							deposit: parseNearAmount(bidAuctionAmount),
+							gas: GAS_FEE,
+						},
+					],
+				})
+			} else {
+				res = await WalletHelper.signAndSendTransaction({
+					receiverId: process.env.MARKETPLACE_CONTRACT_ID,
+					actions: [
+						{
+							methodName: 'storage_deposit',
+							args: depositParams,
+							deposit: STORAGE_ADD_MARKET_FEE,
+							gas: GAS_FEE,
+						},
+						{
+							methodName: 'add_bid',
+							args: params,
+							deposit: parseNearAmount(bidAuctionAmount),
+							gas: GAS_FEE,
+						},
+					],
+				})
+			}
+			if (res?.response.error) {
+				toast.show({
+					text: (
+						<div className="font-semibold text-center text-sm">
+							{res?.response.error.kind.ExecutionError}
+						</div>
+					),
+					type: 'error',
+					duration: 2500,
+				})
+			} else if (res) {
+				onClose()
+				setTransactionRes(res?.response)
+				onSuccess && onSuccess()
+				toast.show({
+					text: (
+						<div className="font-semibold text-center text-sm">{`Successfully add bid auction`}</div>
+					),
+					type: 'success',
+					duration: 2500,
+				})
+			}
+			setIsBidding(false)
+		} catch (err) {
+			sentryCaptureException(err)
+			setIsBidding(false)
+		}
+	}
+
+	const isCurrentBid = (type) => {
+		let list = []
+		data?.bidder_list?.map((item) => {
+			if (type === 'bidder') list.push(item.bidder)
+			else if (type === 'time') list.push(item.issued_at)
+			else if (type === 'amount') list.push(item.amount)
+		})
+		const currentBid = list.reverse()
+
+		return currentBid[0]
+	}
+
+	const checkNextPriceBid = () => {
+		const currentBid = JSBI.BigInt(
+			data?.bidder_list && data?.bidder_list?.length !== 0
+				? isCurrentBid('amount')
+				: data?.price?.$numberDecimal || data?.price || ''
+		)
+		const multiplebid = JSBI.multiply(JSBI.divide(currentBid, JSBI.BigInt(100)), JSBI.BigInt(5))
```

Comment:
ini mau dikali 5% kan? dibikin gini bisa ga kira2?
```suggestion
		const multiplebid = JSBI.multiply(JSBI.BigInt(currentBid), JSBI.BigInt(0.05))
```

Reply:
ngga bisa hu, karena 0.05 bukan BigInt, basically sama ko hu, cuma bedanya code yg sebelumnya itu di bagi 100 dulu baru dikali 5, tapi mungkin kurang readable aja ya hu, ok thanks hu